fix: hide organizer email for collective events when enabled

RajdeepDs opened the pull request:
## What does this PR do?

Prevent displaying the email of team members in collective events if the `hideOrganizerEmail` setting is enabled.

- Fixes #21319
- Fixes [CAL-5771](https://linear.app/calcom/issue/CAL-5771/hide-all-member-email-from-collective-events-if-the-option-to-hide) (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Visual Demo (For contributors especially)

#### Image Demo:
Before:
![Screenshot 2025-05-15 215854](https://github.com/user-attachments/assets/2aada716-ea03-497e-ae6d-5c41bf619b67)
After:
![Screenshot 2025-05-15 215937](https://github.com/user-attachments/assets/eb84ae07-6f17-46fc-a402-72f1d7c3740a)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

    
<!-- This is an auto-generated description by mrge. -->
---

## Summary by mrge
Hides team member emails in collective events when the hideOrganizerEmail setting is enabled, showing only the main attendee's email. This protects organizer privacy in group bookings.

<!-- End of auto-generated description by mrge. -->

RajdeepDs committed:
fix: hide organizer email for collective events when enabled

Prevent displaying the email of team members in collective events if the `hideOrganizerEmail` setting is enabled, ensuring privacy for non-main attendees.

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

Udit-takkar reviewed (changes requested):
I think there's are some problems in the issue.

1) We should hide all the hosts when `hideOrganizerEmail` is enabled.
2) We should hide the emails in case of all event types and not just collective event type

and now in your PR you are also hiding the attendee's email.  bookingInfo?.attendees in case of collecitve event type contains both host and attendee

RajdeepDs committed:
refactor: hide email of all hosts for all event types

RajdeepDs committed:
Merge branch 'fix-hide-team-emails-collective-events' of https://github.com/RajdeepDs/cal.com into fix-hide-team-emails-collective-events

kart1ka committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

kart1ka reviewed (changes requested):
The issue is not fully resolved.
In case of collective event types, the organizer/hosts emails are still visible in the email notifications.
In case of RR event type, the organizer/hosts emails are not hidden at all (booking success page and email notification).
See the video below:
https://cap.link/yppkht7m4gmaa3x

kart1ka commented:
Making the PR draft until then.

RajdeepDs committed:
fix: update email visibility logic for team members and attendees in booking views

- Adjusted conditions to hide organizer emails for collective and round robin scheduling types.
- Enhanced email visibility handling for team members and attendees based on event settings.

kart1ka committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'main' of github.com:RajdeepDs/cal.com into fix-hide-team-emails-collective-events

RajdeepDs committed:
fix: update main attendee check to handle empty email case in bookings single view

kart1ka committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

kart1ka reviewed (changes requested):
There are still scenarios where the emails are visible. See below.

<img width="1470" height="956" alt="Image" src="https://github.com/user-attachments/assets/ab4612c9-e51c-41c4-81de-f9f3d70c0b72" />


<img width="2878" height="1686" alt="Image" src="https://github.com/user-attachments/assets/61b7c3e6-60de-4369-b171-ef788ff58d3d" />

RajdeepDs committed:
Merge branch 'main' of github.com:RajdeepDs/cal.com into fix-hide-team-emails-collective-events

RajdeepDs commented:
Just wanted to ask a few things:
- In the first image, I saw the app is running on `http://acme.app.local:3000/`. How can I use this and regenerate this issue?
- In the second image, the emails are visible. Should I just hide them or replace them with a placeholder?

anikdhabal committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

kart1ka commented:
> @kart1ka Just wanted to ask a few things:
> 
> * In the first image, I saw the app is running on `http://acme.app.local:3000/`. How can I use this and regenerate this issue?
> * In the second image, the emails are visible. Should I just hide them or replace them with a placeholder?

`acme.cal.local` has nothing to do with the emails being visible. If you want to set it up you can do so by following this [guide](https://github.com/calcom/cal.com/blob/main/packages/features/ee/organizations/README.md).

For the second image, yes the emails should be hidden. You can instead show the names of the hosts/organizers.

anikdhabal committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

anikdhabal commented on packages/emails/src/components/WhoInfo.tsx:
```diff
@@ -36,17 +42,27 @@ export function WhoInfo(props: { calEvent: CalendarEvent; t: TFunction }) {
+              email={shouldHideTeamEmails ? "" : member?.email}
             />
           ))}
+          {props.calEvent.attendees.map((attendee) => {
+            // If attendee is also a team member and shouldHideTeamEmails, hide their email
```
No, why don't we filter the attendees array instead?

RajdeepDs replied:
Okay got it! Yes it will be more cleaner approach.

anikdhabal reviewed (changes requested):
I think there are many places where we need to apply this filter, not just these two files. For eg:- bookinglistItem etc

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'fix-hide-team-emails-collective-events' of github.com:RajdeepDs/cal.com into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'fix-hide-team-emails-collective-events' of github.com:RajdeepDs/cal.com into fix-hide-team-emails-collective-events

RajdeepDs committed:
feat: hide emails when hideOrganizerEmail enabled in BookingListItem

RajdeepDs committed:
feat: add schedulingType to getWho and enhance email visibility logic for team members

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

kart1ka commented:
@RajdeepDs There is a type check failing. Please fix it.

RajdeepDs committed:
fix: type errors

RajdeepDs committed:
Merge branch 'fix-hide-team-emails-collective-events' of github.com:RajdeepDs/cal.com into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

RajdeepDs committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

pallava-joshi committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

pallava-joshi reviewed (approved):
LGTM

dhairyashiil reviewed (changes requested):
According to this comment: [link](https://github.com/calcom/cal.com/pull/21340#pullrequestreview-2865456311), we should hide the organiser’s email address from the booking screen, email notifications, and calendar events for **_"all event types"_**. However, I believe this is only implemented for collective and RR event types.

I tested this locally and found that it works for emails for normal event types, but I can still see the organiser’s email on the booking screen/page.

<img width="2560" height="1440" alt="Screenshot 2025-11-10 at 4 38 23 PM (2)" src="https://github.com/user-attachments/assets/761af458-f373-400c-bc97-b64a44c1cc78" />

.
Please implement this feature for all event types and provide a video demonstrating it for each type: user, team, and organisation. (show booking screen, email notifications, and calendar events for each event type)

pallava-joshi reviewed (changes requested):
please resolve merge conflicts too, marking this draft until

anikdhabal committed:
Merge branch 'main' into fix-hide-team-emails-collective-events

keithwillcode committed:
Merge branch 'main' into fix-hide-team-emails-collective-events
